example/basic/t8_basic.cxx

```diff
@@ -0,0 +1,242 @@
+{
+  t8_cmesh_t          cmesh;
+  switch (dim) {
+  case 1:
+    {
+      cmesh =
+        t8_cmesh_new_hypercube (T8_ECLASS_LINE, sc_MPI_COMM_WORLD, 0, 0, 0);
+      break;
+    }
+  case 2:
+    {
+      cmesh =
+        t8_cmesh_new_hypercube (T8_ECLASS_TRIANGLE, sc_MPI_COMM_WORLD, 0, 0,
+                                0);
+      break;
+    }
+  case 3:
+    {
+      cmesh = t8_cmesh_new_hypercube_hybrid (sc_MPI_COMM_WORLD, 0, 0);
+      break;
+    }
+
+  default:
+
+    SC_ABORT_NOT_REACHED ();
+  }
+  return cmesh;
+}
+
+/* The adapt Callback used in this examples. Every element that has an odd
+ * x-coordinate (in its reference element) wrt to its level is refined up to the maximal refinement level */
+static int
+t8_basic_adapt (t8_forest_t forest, t8_forest_t forest_from,
+                t8_locidx_t which_tree, t8_locidx_t lelement_id,
+                t8_eclass_scheme_c * ts, int num_elements,
+                t8_element_t * elements[])
+{
+  int                 level;
+  double              coords[3] = { 0, 0, 0 };
+  const int           maxlevel = 5;
+  level = ts->t8_element_level (elements[0]);
+  /* Check, if the element is not finer than the maximal refinement level */
+  if (level >= maxlevel) {
+    /* We do not refine if the user maximum level is reached */
+    return 0;
+  }
+  ts->t8_element_vertex_reference_coords (elements[0], 0, coords);
+  /*If the x-coordinate lays between 0.25 and 0.75, we refine the element. */
+  if (0.25 <= coords[0] && coords[0] <= 0.75) {
+    return 1;
+  }
+  /* If the x-coordinate is smaller than 0.25 we coarsen the element. */
+  else if (coords[0] < 0.25) {
+    return -1;
+  }
+  /* Do not refine or coarsen the element. */
+  else {
+    return 0;
+  }
+}
+
+/** This function creates, adaptivly refines, partitions and optionally balances a
+ * forest of the hypercube-mesh. For 1D a line is constructed, for 2D a mesh consisting of 2 triangles
+ * forming a quad and in 3D a hybrid hypercube is created.
+ * \param[in] dim         The dimension of the example
+ * \param[in] do_balance  Option to partition the cmesh. If true (non-zero) the cmesh will be partitioned across the processes.  
+*/
+static void
+t8_basic_hypercube (const int dim, const int do_balance)
+{
+  t8_forest_t         forest, forest_adapt;
+  t8_cmesh_t          cmesh;
+  char                vtuname[BUFSIZ];
+  int                 mpirank, mpiret;
+  const int           uniform_lvl = 2;
+
+  t8_global_productionf ("Contructing %i dimensional hypercube mesh \n", dim);
+  /* Create and save the cmesh */
+  cmesh = t8_basic_create_cmesh (dim);
+
+  mpiret = sc_MPI_Comm_rank (sc_MPI_COMM_WORLD, &mpirank);
+  SC_CHECK_MPI (mpiret);
+
+  snprintf (vtuname, BUFSIZ, "cmesh_basic_%i__dim", dim);
+  if (t8_cmesh_vtk_write_file (cmesh, vtuname, 1.0) == 0) {
+    t8_global_productionf ("Output to %s\n", vtuname);
+  }
+  else {
+    t8_global_productionf ("Error when trying to write cmesh to %s.\n",
```

Comment:
```suggestion
    t8_errorf ("Error when trying to write cmesh to %s.\n",
```

We need to be careful in such situations. 
1) in case of an error, an error logging function should be called
2) the global logging functions only print on the main process. But the writing operation could be succesful on the main process and only fail on another proc. In that case, we would not see the error output when using a global output function.